docs/spec/index.md

```diff
@@ -139,3 +141,120 @@ Because some events may overlap in time, it may not be possible to choose a sing
+| `run_id` | ID referencing `run_events.txt` | Conditionally Required | The run that's either added or removed from this roster. Required if `exception_type` is `1`. Optional if `exception_type` is `2`. If `exception_type` is `2` and `run_id` is not blank, then it must match the Run ID that the roster was scheduled to do on this date according to `rosters.txt` |
+
+### `employee_rosters.txt`
+
+Describes which employees are scheduled to which rosters on which dates.
+
+Primary Key: `(roster_id,start_date)`
+
+| **Field Name** | **Type** | **Required** | **Description** |
+| --- | --- | --- | --- |
+| `employee_id` | ID | Required | |
+| `roster_id` | ID referencing `rosters.txt` | Required | |
+| `start_date` | Date | Required | |
+| `end_date` | Date | Required | |
+
+Each roster can only be assigned to one employee on each date. Employees may be scheduled to more than one roster on the same date.
+
+### `employee_run_dates.txt`
+
+Describes which employees are scheduled to which runs on which dates. If `employee_rosters.txt` is used, then describes exceptions to that schedule.
+
+Primary Key: `*`
+
+| **Field Name** | **Type** | **Required** | **Description** |
+| --- | --- | --- | --- |
+| `employee_id` | ID | Required | |
+| `date` | Date | Required | |
+| `service_id` | ID referencing `run_events.txt` | Required | Part of the Run ID, which is refered to as `(service_id, run_id)`. |
+| `run_id` | ID referencing `run_events.txt` | Required | The run that's either added or removed from this employee's schedule. If `exception_type` is `2` and `run_id` is not blank, then it must match a Run ID that the employee was scheduled to do on this date according to `employee_rosters.txt`, `rosters.txt` and `roster_dates.txt`. |
+| `exception_type` | Enum | Optional | `1` (or blank) - The run is assigned to this employee on the specified date.<br />`2` - The employee will not work this run on this date. |
+
+If a feed doesn't represent rosters, it can still assign employees to runs by putting every run for every date in this file. In that case, the `exception_type` column can be omitted because every row would be adding a date, which is the default when the column is blank.
+
+Each run can only be assigned to one employee on each date. Employees may be scheduled to more than one run on the same date.
+
+### `vehicle_assignments.txt`
+
+Primary Key: `*`
+
+| Field Name | Type | Required | Description |
+|---|---|---|---|
+| `date` | Date | Required |  |
+| `service_id` | ID referencing `calendar.service_id` or `calendar_dates.service_id` | Optional | Identifies a set of dates when the run is scheduled to take place. Required if `block_id`s are repeated between different `service_id`s. |
+| `block_id`		  | ID referencing `trips.block_id` | Conditionally required | Identifies the block. Either `trip_id` or `block_id` must be specified. |
```

Comment:
Block could be always required if:
- We say you can only use this feature if you use block IDs (they're optional in [trips.txt](https://gtfs.org/documentation/schedule/reference/#tripstxt))
- Either:
    - You have to include the block, even if you include the trip. (And the block must be the same block as the trip's block.)
    - trip_id is removed (see previous comment).

Alternatively, it would also be possible to say that exactly one of block_id and trip_id is required, which might be less error-prone than allowing both and ignoring the block if the trip exists.

Reply:
Let's explore removing `vehicle_assignments`.`trip_id`

Reply:
f3281fe536c40490a0fb9a221c7cfd1f41b64e15